### PR TITLE
i#2101 : adding some hint nops to x86 decoding

### PR DESCRIPTION
**Pull request by Simorfo**
Fixes issue 2101

I did not know how to make good test case out of it.
With this modification of common-decode, we need to run dynamorio with a client in order to get a different behavior between the current version and the patched one (being the one which is transparent)

### CODE REVIEW

**Comment by derekbruening** [comment]
Separate opcodes are tested in all different ways (decode, instr_create, encode) in the api/ir* tests but these are unusual in that they're not only the same opcode but not even on the encoding list so there's no way they'll be generated from IR.  If they're not in the dis-udis86-randtest.raw binary file they should probably be added as a special test inside api/ir_x86.c that verifies that each raw encoding is decoded to nop_modrm.

**Comment by Simorfo** [comment]
No they're not in the dis-udis86-randtest.raw binary file.
Is this test case better ?
I would have rather had a test where dynamorio behaves differently than the program on that opcode...

File: suite/tests/api/ir_x86.c [hunk 1]
```diff
@@ -737,6 +737,22 @@ test_nop_xchg(void *dc)
...
 
+static void
+test_hint_nops(void *dc)
+{
+    byte *pc;
+    instr_t *instr;
+    /* ensure we treat as nop. */
+    buf[0] = 0x0f;
+    buf[1] = 0x18;
+    buf[2] = 0x38;
```
Comment (derekbruening) [inline suite/tests/api/ir_x86.c]: How about an array of these { { 0x0f, 0x18, 0x38 }, {...}} to cover all 4 variants?

File: suite/tests/api/ir_x86.c [hunk 2]
```diff
@@ -737,6 +737,22 @@ test_nop_xchg(void *dc)
...
+    instr = instr_create(dc);
+    pc = decode(dc, buf, instr);
+    ASSERT(instr_get_opcode(instr) == OP_nop_modrm);
+    instr_destroy(dc, instr);
+
```
Comment (derekbruening) [inline suite/tests/api/ir_x86.c]: nit: extraneous line

**Comment by derekbruening** [comment]
Actually maybe we could add the other 0f1[9-f] ones to the test too if that's easy.

**Comment by Simorfo** [comment]
Here is the new test.
By the way, some online disassemblers give something different than nop for 0f1a : bndldx and 0f1b : bndstx from Intel MPX (Memory Protection Extensions)

**Comment by derekbruening** [comment]
Adding MPX is #1312.

**Pull request merged**